runelite-client/src/main/java/net/runelite/client/game/chatbox/ChatboxItemSearch.java

```diff
@@ -0,0 +1,306 @@
+import net.runelite.api.widgets.ItemQuantityMode;
+import net.runelite.api.widgets.JavaScriptCallback;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetPositionMode;
+import net.runelite.api.widgets.WidgetSizeMode;
+import net.runelite.api.widgets.WidgetTextAlignment;
+import net.runelite.api.widgets.WidgetType;
+import net.runelite.client.callback.ClientThread;
+import net.runelite.client.game.ItemManager;
+
+public class ChatboxItemSearch extends ChatboxTextInput
+{
+	private static final int ICON_HEIGHT = 32;
+	private static final int ICON_WIDTH = 36;
+	private static final int PADDING = 6;
+	private static final int MAX_RESULTS = 24;
+	private static final int FONT_SIZE = 16;
+	private static final int HOVERED_OPACITY = 128;
+
+	private final ChatboxPanelManager chatboxPanelManager;
+	private final ItemManager itemManager;
+	private final Client client;
+
+	private HashMap<Integer, ItemComposition> results = new LinkedHashMap<>();
+	private String tooltipText;
+	private int index = -1;
+
+	@Getter
+	private Consumer<Integer> onItemSelected;
+
+	@Inject
+	public ChatboxItemSearch(ChatboxPanelManager chatboxPanelManager, ClientThread clientThread,
+		ItemManager itemManager, Client client)
+	{
+		super(chatboxPanelManager, clientThread);
+		this.chatboxPanelManager = chatboxPanelManager;
+		this.itemManager = itemManager;
+		this.client = client;
+
+		lines(1);
+		prompt("Item Search");
+		onChanged(searchString ->
+			clientThread.invokeLater(() ->
+			{
+				filterResults();
+				update();
+			}));
+	}
+
+	@Override
+	protected void update()
+	{
+		Widget container = chatboxPanelManager.getContainerWidget();
+		container.deleteAllChildren();
+
+		Widget promptWidget = container.createChild(-1, WidgetType.TEXT);
+		promptWidget.setText(getPrompt());
+		promptWidget.setTextColor(0x800000);
+		promptWidget.setFontId(getFontID());
+		promptWidget.setOriginalX(0);
+		promptWidget.setOriginalY(5);
+		promptWidget.setXPositionMode(WidgetPositionMode.ABSOLUTE_CENTER);
+		promptWidget.setYPositionMode(WidgetPositionMode.ABSOLUTE_TOP);
+		promptWidget.setOriginalHeight(FONT_SIZE);
+		promptWidget.setXTextAlignment(WidgetTextAlignment.CENTER);
+		promptWidget.setYTextAlignment(WidgetTextAlignment.CENTER);
+		promptWidget.setWidthMode(WidgetSizeMode.MINUS);
+		promptWidget.revalidate();
+
+		buildEdit(0, 5 + FONT_SIZE, container.getWidth(), FONT_SIZE);
+
+		Widget separator = container.createChild(-1, WidgetType.LINE);
+		separator.setOriginalX(0);
+		separator.setOriginalY(8 + (FONT_SIZE * 2));
+		separator.setXPositionMode(WidgetPositionMode.ABSOLUTE_CENTER);
+		separator.setYPositionMode(WidgetPositionMode.ABSOLUTE_TOP);
+		separator.setOriginalHeight(0);
+		separator.setOriginalWidth(16);
+		separator.setWidthMode(WidgetSizeMode.MINUS);
+		separator.setTextColor(0x666666);
+		separator.revalidate();
+
+		int x = PADDING;
+		int y = PADDING * 3;
+		for (int i = 0; i < results.size(); i++)
```

Comment:
This should probably iterate the keyset as it can have holes in it due to not all items matching the search criteria.